modules/tensor_mechanics/src/materials/RadialReturnStressUpdate.C

```diff
@@ -108,6 +111,33 @@ RadialReturnStressUpdate::updateState(RankTwoTensor & strain_increment,
    * currently just the elasticity tensor, irrespective of compute_full_tangent_operator
    */
   tangent_operator = elasticity_tensor;
+
```

Comment:
Because of where this is, this will get used for both creep and plasticity. Have you established that we need to follow different procedures for the two cases?  Maybe it is the right thing to do for both.  I'd make a  simple test and see what it does on a plasticity problem.

If it's not the right thing to do in both cases, we need this base class to know whether it's dealing with a creep or plasticity model (it kind of already does) and choose appropriately.

Reply:
Yeah, this is why the tests are failing for `strain_energy_density/inc_model_elas_plas`. Stress_xx at the last timestep is 9.798429e-11 nominally, 1.922578e-10 with the tangent operator...I'm surprised it is changing at all with the tangent.

@naveen-pr has the formulation for creep+plasticity and plasticity, which is fundamentally different from just creep. I'm not sure how to separate plasticity models from creep models though. A quick look through `IsotropicPlasticityStressUpdate` and `PowerLawCreepStressUpdate` both inherit from `RadialReturnStressUpdate`, and don't have a defining feature I see that would set which tangent modulus to calculate. Do you have an idea of how to separate the two?

Reply:
The tangent operators need to combined in a smart way in `ComputeMultipleInelasticStress` as well.

Reply:
@naveen-pr have you established how to combine tangents yet?

Reply:
Not yet @tophmatthews , I'm looking into it

Reply:
So there is one for plasticity, and one for no plasticity. What about volumetric swelling? Will they be different? I guess that depends on how they combine.

Reply:
Volumetric swelling in your model depends on the hydrostatic stress, so there will be an invert involved again, but that should be 'separable'/'combinable' with the tangent for creep (I think). For only creep without plasticity I mean.

Reply:
Regarding that broken test, that's probably just because it needs to have the floor made a little bigger on that variable -- it's essentially zero, and this just made it a little different.

Reply:
We are currently passing a string `inelastic_strain_name` into the constructor of `RadialReturnStressUpdate` that is currently either `creep` or `plastic`. We don't want to do string comparisons in the computation, but we could pass an enumeration or something down instead of a string, and then select the string for those property names based on the value of that enumeration, and then also use that enumeration to select the tangent operator.

Reply:
Even if we only get this working initially for one inelastic model, and not for the combined case, that would be a big step forward, and I would be fine with that (as long as we generate an error if you try to use it in the combined case).

Reply:
I don't think the creep formulation is limited by power law here. The only limitation is that the stress form used to calculate the strain for creep looks like:
```
stress_delta = effective_trial_stress - _three_shear_modulus * scalar
```
This is because I hijack the newton solve which is dstrain/dscalar in the tangent operator calculation.

Reply:
@tophmatthews Well no, not in the current formulation of the tangent (Eqn. 87 in the document I sent out is general but implicitly assuming a time hardening form of creep). I'm not sure what the tangent would look like if creep is assumed to be in strain hardening form, for example.

Reply:
@bwspenc So I'm thinking the best way to handle combining the tangent modulus in non-plastic regimes is to add another enum to `tangent_operator = stress_dependent`. Then the tangent operator calculated in the creep, and other stress dependent inelastic strain formulations (as opposed to strain dependent models like plasticity) will return the non inverted tangent (A and B below):
```
I^sym:dstress + A:dstress + B:dstress = C:dstrain
```
Then, `ComputeMultipleInelasticStress::computeQpJacobianMult()` can combine them and do a single inversion.
```
dstress = (I^sym + A + B)^-1 C:dstrain
```

The alternative is to do a lot of manipulation in `computeQpJacobianMult`, requiring inverted rank four tensors to be inverted back, multiplied against a bunch each other a bunch etc...it'll get costly if many models are to be combined.

The default `tangent_operator=nonlinear` can be retained, so plastic models aren't affected. This may help nudge things along with creep+plasticity regardless, but it can remain an option. The user will specifically have to choose `tangent_operator=stress_dependent` to calculate the tangent, and have the tangents combined in this way. The flag in updateState might have to be updated to an enum instead of a bool though.

Reply:
@tophmatthews I still haven't had a chance to really wrap my head around this, but I definitely prefer something along the lines of your first option above rather than doing multiple inversions of rank four tensors.

As far as the code architecture goes, we could add another level of inheritance, with intermediate classes for creep and plasticity wedged between RadialReturnStressUpdate and the individual models. I'm a little hesitant to suggest this because too many levels of inheritance gets confusing, but it might be the best solution, rather than having logic to switch between behaviors for the two cases in RadialReturnStressUpdate.

Reply:
Inheritance may be helpful, but there'll still need to be a check to see if any plastic models are present, or rather strain-dependent models. If there are none, then the stress-dependent models can be combined, inverted, and then multiplied against the modulus. If they are...well then it'll have to be a (I + X + Y)C type combination. I think an additional enum to `tangent_operator` looks like the way to go, but I'm not positive. I 'll try and implement and then we can wee

Reply:
One thing that is important before all these changes is that you agree with the formulation in @naveen-pr 's document I sent earlier. If we're good on that math, then implementation is a different beast.

Reply:
The combining is done in `ComputeMultipleInelasticStress`, and will need to be re-worked.  We probably shouldn't be getting the consistent tangent operator, but instead getting the X and Y matrices from the individual models.

Reply:
Yeah exactly!